api.py

```diff
@@ -367,7 +388,134 @@ def set_generic_data(dataset_id, key, data_string, **kwargs):
+
+    coord['z'] = 0
+
+    return coord
+
+
+def convert_samples_to_entity_set(sample_list):
+    conduce_keys = ['id', 'kind', 'time', 'point', 'path', 'polygon']
+    entities = []
+    for sample in sample_list:
+        if not isinstance(sample['time'], datetime):
+            raise TypeError('time must be a datetime object', sample['time'])
+        coordinates = []
+        if 'point' in sample:
+            coordinates = [convert_coordinates(sample['point'])]
+        if 'path' in sample:
+            if len(coordinates) > 0:
+                raise KeyError('A sample may only contain one of point, path or polygon', sample)
+            if len(sample['path']) < 2:
+                raise KeyError('paths must have at least two points')
+            for point in sample['path']:
+                coordinates.append(convert_coordinates(point))
+        if 'polygon' in sample:
+            if len(coordinates) > 0:
+                raise KeyError('A sample may only contain one of point, path or polygon', sample)
+            if len(sample['polygon']) < 3:
+                raise KeyError('polygons must have at least three points')
+            for point in sample['polygon']:
+                coordinates.append(convert_coordinates(point))
+
+        sample['_kind'] = sample['kind']
+        attribute_keys = [key for key in sample.keys() if key not in conduce_keys]
+        attributes = util.get_attributes(attribute_keys, sample)
+
+        entity = {
+            'identity': sample['id'],
+            'kind': sample['kind'],
+            'timestamp_ms': util.datetime_to_timestamp_ms(sample['time']),
+            'endtime_ms': util.datetime_to_timestamp_ms(sample['time']),
+            'path': coordinates,
+            'attrs': attributes
+        }
+
+        entities.append(entity)
+
+    return {'entities': entities}
+
+
+def ingest_samples(dataset_id, sample_list, **kwargs):
+    """
+    Upload a :ref:`sample list <conduce-entities>` to the Conduce datastore.
+
+    A convenience method that uploads an list of entity samples to the Conduce datastore and waits for the job to complete. This function POSTs a sample list to Conduce and :py:func:`wait_for_job` until the ingest job completes.
+
+    Parameters
+    ----------
+    dataset_id : string
+        The UUID that identifies the dataset to modify.
+    sample_list : list
+        A list of entity samples.  See :ref:`conduce-entities` for documentation on how to build a sample list.
+
+    **kwargs:
+        See :py:func:`make_post_request`
+
+    Returns
+    -------
+    requests.Response
+        Returns an error or the final response message when the job is no longer running.
+
+    Raises
+    ------
+    requests.HTTPError
+        Requests that result in an error raise an exception with information about the failure. See :py:func:`requests.Response.raise_for_status` for more information.
+    """
+
+    if not isinstance(sample_list, list):
```

Comment:
Hmm, that does seem weird.

I did it so that a user could ingest a single sample without adding it to a list.  Maybe it's not worth it.

Reply:
Consider the use case where a user wants to ingest a single sample.  It seems arduous to make the user but the sample in a list.  The purpose is to provide convenience to the user.  If they have multiple samples, they obviously need to put them in a list.  If they have one sample don't make them put brackets around it, to them the input is not a list.

Reply:
I'll cave on this one.  If it's going to behave the way I coded it, the documentation should indicate as much.  It seems complicated to document that a function parameter can be one of many types.  However, I'd say that's the way python intends things to work.

If I were writing this in C++ I would override the function such that the user could pass a single struct or a list of structs.

Reply:
I think it's fine to support two types in the same argument, compiled languages don't support it but it's fine in scripting languages.  Typescript supports the or operator so a function can accept multiple type or return multiple types, no issues.  

We should update the parameter name to be different than sample_list (since we support a single sample), update the header documentation for awarness and add a small line describing two supported types so when a developer is skimming through the code, it doesn't throw him off

Reply:
This conditions throws an error now.